app/controllers/BaseNavigationController.scala

```diff
@@ -59,10 +63,24 @@ trait BaseNavigationController extends BaseController with Logging {
       } yield updatedAnswers
     }
 
-  private def save[A](page: QuestionPage[A], answer: A)
-                     (implicit request: DataRequest[_], format: Format[A]): Future[UserAnswers] =
-    save(page, answer, request.userAnswers)
-
+  private[controllers] def markErrorAsFixedIfPresent(page: QuestionPage[_])(implicit request: DataRequest[_]): UserAnswers = {
+    if(request.userAnswers.haveAllSubmissionErrorsBeenFixed) {
+     request.userAnswers
+    } else {
+      val indexOfError = Try(page match {
+        case LocalReferenceNumberPage(_) => request.userAnswers.submissionFailures.indexWhere(_.errorType == localReferenceNumberError)
```

Comment:
Is it possible for .indexWhere to return -1 here? If so should we take that in to account in?